cookbooks/particle_injection.prm

```diff
@@ -0,0 +1,247 @@
+# these boundaries; this is also often used to specify symmetry boundaries).
+
+subsection Model settings
+  set Fixed temperature boundary indicators   = bottom, top # left, right 
+
+  # The next parameters then describe on which parts of the
+  # boundary we prescribe a zero or nonzero velocity and
+  # on which parts the flow is allowed to be tangential.
+  # Here, all four sides of the box allow tangential
+  # unrestricted flow but with a zero normal component:
+
+  set Zero velocity boundary indicators       =
+  set Prescribed velocity boundary indicators =
+  set Tangential velocity boundary indicators = bottom, top, left, right
+
+  # The final part of this section describes whether we
+  # want to include adiabatic heating (from a small
+  # compressibility of the medium) or from shear friction,
+  # as well as the rate of internal heating. We do not
+  # want to use any of these options here:
+
+  set Include adiabatic heating               = false
+  set Include shear heating                   = false
+end
+
+
+# The following two sections describe first the
+# direction (vertical) and magnitude of gravity and the
+# material model (i.e., density, viscosity, etc). We have
+# discussed the settings used here in the introduction to
+# this cookbook in the manual already.
+
+# HL: A Rayleigh number of 1E5
+subsection Gravity model
+  set Model name = vertical
+
+  subsection Vertical
+    set Magnitude = 10   # = Ra / Thermal expansion coefficient
+  end
+end
+
+
+subsection Material model
+  set Model name = simple # default:
+
+  subsection Simple model
+    set Reference density             = 1
+    set Reference specific heat       = 1
+    set Reference temperature         = 0
+    set Thermal conductivity          = 1
+    set Thermal expansion coefficient = 1
+    set Viscosity                     = 1E-3
+  end
+end
+
+
+# We here use a globally refined mesh without
+# adaptive mesh refinement.
+
+subsection Mesh refinement
+  set Initial global refinement          = 0
+  set Initial adaptive refinement        = 0
+  set Time steps between mesh refinement = 0
+end
+
+
+# The final part is to specify what ASPECT should do with the
+# solution once computed at the end of every time step. The
+# process of evaluating the solution is called `postprocessing'
+# and we choose to compute velocity and temperature statistics,
+# statistics about the heat flux through the boundaries of the
+# domain, and to generate graphical output files for later
+# visualization. 
+
+subsection Postprocess
+   set List of postprocessors = basic statistics, visualization, tracers, heat flux statistics, velocity statistics
+#  set List of postprocessors = basic statistics, visualization, heat flux statistics, velocity statistics
+#  set List of postprocessors = basic statistics, heat flux statistics, velocity statistics
+#  set List of postprocessors = basic statistics, heat flux statistics, velocity statistics, tracers
```

Comment:
A test really only needs to run a particular piece of code in a way that you can check once, when you write the test, that it does the correct thing and so that if it is broken, it produces different output. The testcase doesn't have to realistic at all; it just needs to produce different output if someone changes the code in incorrect ways.

Many of the tests we have work on 2x2 grids or something similarly pointless. But that's good enough for the purpose of a test :-)